### PR TITLE
docs(python): update docs

### LINKED ISSUE #373: Missing Contribution guideline file
# Description

I see that your repo is missing its Contribution guideline file. Adding a Contribution file improves your community profile
Cant I propose a Contribution file?

### LINKED ISSUE #715: Python write_deltalake: when will this be production ready?
# Description
Just wondering when python write_deltalake will be ready for production use as it says its still experimental
 
**Use Case**
We need this for use from an Azure App Service as it will save us having to deploy multiple reources

**Related Issue(s)**

### PR DESCRIPTION
**Pull request by wjones127**
# Description
The description of the main changes of your pull request

# Related Issue(s)

- closes #715
- closes #373

# Documentation

### CODE REVIEW

**Comment by roeap** [comment]
Just went through our issues, and I guess we can close #373, once this is merged.

File: python/tests/test_writer.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def test_partition_overwrite(
         tmp_path,
         sample_data,
         mode="overwrite",
-        partitions_filters=[("p1", "=", "1")],
+        partition_filters=[("p1", "=", "1")],
```
Comment (wjones127) [inline python/tests/test_writer.py:588]: We haven't yet released with this feature, so this API change should be fine.

File: python/deltalake/writer.py [hunk 2]
```diff
@@ -85,15 +85,15 @@ def write_deltalake(
...
 ) -> None:
-    """Write to a Delta Lake table (Experimental)
+    """Write to a Delta Lake table
 
     If the table does not already exist, it will be created.
 
-    This function only supports protocol version 1 currently. If an attempting
-    to write to an existing table with a higher min_writer_version, this
-    function will throw DeltaTableProtocolError.
+    This function only supports writer protocol version 2 currently. If an
```
Comment (roeap) [inline python/deltalake/writer.py:94]:
```suggestion
    This function only supports writer protocol version 2 currently. When
```

File: python/docs/source/usage.rst [hunk 3]
```diff
@@ -378,10 +406,6 @@ Writing Delta Tables
 
 .. py:currentmodule:: deltalake
 
-.. warning::
-    The writer is currently *experimental*. Please use on test data first, not
-    on production data. Report any issues at https://github.com/delta-io/delta-rs/issues.
-
```
Comment (roeap) [inline python/docs/source/usage.rst:384]: 🎉

**Review by roeap — APPROVED** [review]
Excellent update!

**Review by roeap — APPROVED** [review]

**Pull request merged**